src/stores/featureFlagsStore.ts

```diff
@@ -0,0 +1,139 @@
+  const clientFlags = ref<Record<string, unknown>>({})
+
+  // Flag to indicate if the store has received initial server flags
+  const isReady = ref(false)
+
+  // Track update version to prevent race conditions
+  let updateVersion = 0
+
+  /**
+   * Update server feature flags
+   * Called when WebSocket receives feature_flags message
+   */
+  function updateServerFlags(flags: Record<string, unknown>) {
+    // Validate input
+    if (!flags || typeof flags !== 'object') {
+      console.error('[FeatureFlags] Invalid flags received:', flags)
+      return
+    }
+
+    // Increment version for this update
+    const currentVersion = ++updateVersion
+
+    // Use nextTick to batch updates and check version
+    void nextTick(() => {
+      // Only apply if this is still the latest update
+      if (currentVersion === updateVersion) {
+        serverFlags.value = { ...flags }
+        isReady.value = true
+        console.log('[FeatureFlags] Server flags updated:', serverFlags.value)
+      } else {
+        console.log(
+          '[FeatureFlags] Skipping outdated update, version:',
+          currentVersion
+        )
+      }
+    })
+  }
+
+  /**
+   * Update client feature flags
+   */
+  function updateClientFlags(flags: Record<string, unknown>) {
+    clientFlags.value = { ...flags }
+    console.log('[FeatureFlags] Client flags updated:', clientFlags.value)
+  }
+
+  /**
+   * Get a server feature flag value (reactive)
+   */
+  function getServerFeature<T = unknown>(
+    featurePath: string,
+    defaultValue?: T
+  ): T {
+    return get(serverFlags.value, featurePath, defaultValue) as T
+  }
+
+  /**
+   * Check if server supports a feature (reactive)
+   */
+  function serverSupportsFeature(featurePath: string): boolean {
+    return get(serverFlags.value, featurePath) === true
+  }
+
+  /**
+   * Get a client feature flag value (reactive)
+   */
+  function getClientFeature<T = unknown>(
+    featurePath: string,
+    defaultValue?: T
+  ): T {
+    return get(clientFlags.value, featurePath, defaultValue) as T
+  }
+
+  /**
+   * Computed property for manager v4 support
+   */
+  const supportsManagerV4 = computed<boolean>(() => {
+    return (
+      getServerFeature<boolean>('extension.manager.supports_v4', false) === true
+    )
+  })
+
+  /**
+   * Computed property for client manager v4 UI support
+   */
+  const clientSupportsManagerV4UI = computed<boolean>(() => {
+    return getClientFeature<boolean>('supports_manager_v4_ui', false) === true
+  })
+
+  /**
+   * Reset store state (useful for reconnection)
+   */
+  function resetStore() {
+    console.log('[FeatureFlags] Resetting store state')
+    serverFlags.value = {}
+    isReady.value = false
+    // Note: We don't reset clientFlags as they're local
+    // Increment version to invalidate any pending updates
+    updateVersion++
+  }
+
+  /**
+   * Cleanup store (for unmounting or preventing memory leaks)
+   */
+  function cleanup() {
```

Comment:
Why is this exposed?
Wouldn't you want to control this to prevent arbitrary components from accidentally clearing state that others might depend on?